### PR TITLE
[Bugfix][Sampling] Clear empty side on thinking-budget asymmetric SWAP

### PR DESCRIPTION
**Pull request by hsusul**
## Purpose

Fix a production bug in `ThinkingBudgetStateHolder.sync_batch` where swapping a budgeted request with an unbudgeted batch slot leaves stale thinking-budget state at the empty index.

When the scheduler/input-batch reorders requests (`MoveDirectionality.SWAP`), a budgeted request at index `i` swapped with an unbudgeted request at index `j` previously:

1. Copied state from `i` → `j` via `dict.get()` + assign
2. Left the original entry at `i` intact

Both indices then referenced the **same** state object. Subsequent forcing of end-of-thinking tokens could therefore apply to the wrong request in mixed `thinking_token_budget` / normal batches.

This matches the correct pop-based pattern already used by logits processors (`builtin.py`) and `swap_dict_values`.

**Root cause:** asymmetric dict SWAP used `.get()` instead of `.pop()`, so the empty side was never cleared.

**Solution:** `pop` both indices before reassigning, identical to neighboring batch-state movers.

### Why this is not a duplicate

Searched open/closed issues, open/merged PRs, and recent commits for:

- `ThinkingBudgetStateHolder`
- `thinking_budget_state` / asymmetric SWAP / stale thinking budget
- `sync_batch` SWAP

Related open PRs (`#44690`, `#44812`, `#43210`, `#41768`, etc.) address tool-call / end-token / re-entry behavior, not batch-index SWAP bookkeeping. No existing issue or PR covers this defect.

AI assistance was used; I reviewed every changed line and ran the tests below.

## Test Plan

```bash
.venv/bin/python -m pytest tests/v1/sample/test_thinking_budget_state.py -v
.venv/bin/pre-commit run --files \
  vllm/v1/sample/thinking_budget_state.py \
  tests/v1/sample/test_thinking_budget_state.py
git diff --check
```

Regression covers:

- budgeted ↔ unbudgeted SWAP (both directions)
- budgeted ↔ budgeted SWAP (existing contract preserved)

## Test Result

**Before fix** (regression fails):

```text
assert [0, 1] == [1]
# stale key left at empty side; both keys share one state object
```

**After fix:**

```text
2 passed
```

Pre-commit hooks on changed files: passed (ruff, mypy-3.10, SPDX, etc.).
`git diff --check`: clean.

## Risk Assessment

- **Scope:** two-line change in SWAP branch of `sync_batch`; no API or config changes.
- **Risk:** low. Behavior for two-budgeted and unidirectional moves is unchanged; only the previously incorrect empty-side case changes.
- **Blast radius:** reasoning models using per-request `thinking_token_budget` under batch reordering.

Made with [Cursor](https://cursor.com)

### CODE REVIEW

**Review by njhill — APPROVED** [review]
Thanks @hsusul
